segments_config/validators.py

```
@@ -0,0 +1,80 @@
+import re
+
+from io import StringIO
+
+from django.core import validators
+from django.core.exceptions import ValidationError
+from string_utils import is_url
+
+
+def url_regex(url):
+    retval = is_url(url)
+    if retval is False:
+        raise ValidationError("Enter Valid URL")
+
+alphanum = validators.RegexValidator(
+    regex=r'^[A-Za-z0-9]+$',
+    message='Only alphanumeric characters are allowed.'
+)
+
+_rule_name_regex = re.compile(
+    r'^@[0-9a-zA-Z]+$', re.I
+)
+
+_rule_pattern_regex = re.compile(
+    r'^[^\s][0-9A-z,/\*_\-\.]+$', re.I
+)
+
+
+def _rule_name(key):
+    assert key is not None
+    return _rule_name_regex.match(key)
+
+
+def _rule_pattern(pattern):
+    assert pattern is not None
+    return _rule_pattern_regex.match(pattern)
+
+
+def _rules_key_value(v):
+    assert v is not None
+
+    keys = []
+    item = key = value = None
+
+    like_file = StringIO(v)
+    content = iter(like_file.readlines())
+
+    for item in content:
+        item = item.strip()
+        if not item:
+            continue
+
+        if key is None:
+            key = item
+            if not _rule_name(key):
+                return "{} is not allowed like rule name".format(key)
```

raise errors instead of returning strings